pkg/controller/daemon/util/daemonset_util.go

```diff
@@ -144,6 +144,10 @@ func SurgeCount(ds *apps.DaemonSet, numberToSchedule int) (int, error) {
 	if r == nil {
 		return 0, nil
 	}
+	// If surge is not requested, we should default to 0.
+	if r.MaxSurge == nil {
```

Comment:
Can you add a unit test for this?

Reply:
Sure. Modified the existing unit to run with/without featuregate flag enabled. As discussed, I'll open a follow-up PR to modify all the tests.